k4FWCore/components/PodioInput.cpp

```diff
@@ -85,8 +92,13 @@ void PodioInput::fillReaders() {
   m_readers["edm4hep::ClusterCollection"] = [&](std::string_view collName) {
     maybeRead<edm4hep::ClusterCollection>(collName);
   };
+  m_readers["edm4hep::TrackerHit3DCollection"] = [&](std::string_view collName) {
+    maybeRead<edm4hep::TrackerHit3DCollection>(collName);
+  };
   m_readers["edm4hep::TrackerHitCollection"] = [&](std::string_view collName) {
-    maybeRead<edm4hep::TrackerHitCollection>(collName);
+    warning() << "Reading a TrackerHitCollection via TrackerHit3DCollection (" << collName
```

Comment:
I'm not sure about this warning; the only way you can reach here is when TrackerHitCollection is available (otherwise the type will never be `edm4hep::TrackerHitCollection`) in which case `TrackerHit3DCollection` is an alias to `TrackerHitCollection` meaning this warning wouldn't be true, no?

Reply:
It will be possible to come here from old files, I think. In that case the old name will still be in the file metadata and propagate until here. However, we are fine just reading that through the new type. The warning might not be necessary though, as it will probably do the right thing. Additionally, all the other changes that we are currently introducing will break existing files in any case.

Reply:
How are old files going to be readable? I'm quite sure they are not. ROOT finds something called "TrackerHitCollection" that it doesn't know anything about and it's impossible for it to reconstruct a "Tracker3DHitCollection". So this is why I think this warning can't be triggered because if the collections are saved as `TrackerHitCollection` then they can't be read